app/src/main/java/com/infomaniak/mail/data/UiSettings.kt

```diff
@@ -0,0 +1,240 @@
+/*
+ * Infomaniak kMail - Android
+ * Copyright (C) 2022 Infomaniak Network SA
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.infomaniak.mail.data
+
+import android.content.Context
+import androidx.annotation.ColorRes
+import androidx.annotation.IdRes
+import androidx.annotation.StringRes
+import androidx.annotation.StyleRes
+import androidx.appcompat.app.AppCompatDelegate
+import com.infomaniak.lib.core.utils.Utils.enumValueOfOrNull
+import com.infomaniak.lib.core.utils.transaction
+import com.infomaniak.mail.R
+import kotlin.reflect.KMutableProperty0
+
+class UiSettings private constructor(context: Context) {
+
+    private val sharedPreferences = context.applicationContext.getSharedPreferences("UISettings", Context.MODE_PRIVATE)
+
+    fun removeUiSettings() = sharedPreferences.transaction { clear() }
+
+    //region Thread density
+    private var _threadDensity: String?
+        get() = getPrivateSetting(::threadDensity, DEFAULT_THREAD_DENSITY)
+        set(value) = sharedPreferences.transaction { putString(::threadDensity.name, value) }
```

Comment:
The same way there's a function `getPrivateSetting` couldn't there be a function `setPrivateSetting` or something that factorises this code which is called multiple times ?